pkg/apis/internal.admin.acorn.io/v1/resources_test.go

```diff
@@ -0,0 +1,257 @@
+				Apps:          1,
+				VolumeStorage: resource.MustParse("1Mi"),
+			},
+		},
+	}
+
+	// Run the test cases
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.current.Add(tc.incoming)
+			tc.current.Equals(tc.expected)
+		})
+	}
+}
+
+func TestRemove(t *testing.T) {
+	// Define test cases
+	testCases := []struct {
+		name     string
+		current  Resources
+		incoming Resources
+		all      bool
+		expected Resources
+	}{
+		{
+			name:    "remove from empty resources",
+			current: Resources{},
+			incoming: Resources{
+				Apps:          1,
+				VolumeStorage: resource.MustParse("1Mi"),
+			},
+			expected: Resources{},
+		},
+		{
+			name: "remove from existing resources",
+			current: Resources{
+				Apps:          1,
+				VolumeStorage: resource.MustParse("1Mi"),
+			},
+			incoming: Resources{
+				Apps:          1,
+				VolumeStorage: resource.MustParse("1Mi"),
+			},
+			expected: Resources{},
+		},
+		{
+			name: "remove persistent counts with all",
+			current: Resources{
+				Secrets: 1,
+			},
+			incoming: Resources{
+				Secrets: 1,
+			},
+			all:      true,
+			expected: Resources{},
+		},
+		{
+			name: "does not remove persistent counts without all",
+			current: Resources{
+				Secrets: 1,
+			},
+			incoming: Resources{
+				Secrets: 1,
+			},
+			expected: Resources{
+				Secrets: 1,
+			},
+		},
+	}
+
+	// Run the test cases
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.current.Remove(tc.incoming, tc.all)
+			tc.current.Equals(tc.expected)
+		})
+	}
+}
+
+func TestEquals(t *testing.T) {
+	// Define test cases
+	testCases := []struct {
+		name     string
+		current  Resources
+		incoming Resources
+		expected bool
+	}{
+		{
+			name:     "empty resources",
+			current:  Resources{},
+			incoming: Resources{},
+			expected: true,
+		},
+		{
+			name: "equal resources",
+			current: Resources{
+				Apps:          1,
+				Secrets:       1,
+				VolumeStorage: resource.MustParse("1Mi"),
+			},
+			incoming: Resources{
+				Apps:          1,
+				Secrets:       1,
+				VolumeStorage: resource.MustParse("1Mi"),
+			},
+			expected: true,
+		},
+		{
+			name: "unequal resources",
+			current: Resources{
+				Apps:          1,
+				Secrets:       1,
+				VolumeStorage: resource.MustParse("1Mi"),
+			},
+			incoming: Resources{
+				Apps:          2,
+				Secrets:       2,
+				VolumeStorage: resource.MustParse("1Mi"),
+			},
+			expected: false,
+		},
+	}
+
+	// Run the test cases
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.current.Equals(tc.incoming) != tc.expected {
+				t.Errorf("expected %v, got %v", tc.expected, !tc.expected)
+			}
+		})
+	}
+}
+
+func TestFits(t *testing.T) {
+	// Define test cases
+	testCases := []struct {
+		name        string
+		current     Resources
+		incoming    Resources
+		expectedErr error
+	}{
+		{
+			name:     "empty resources",
+			current:  Resources{},
+			incoming: Resources{},
+		},
```

Comment:
This specific case was causing the weird error you were seeing @StrongMonkey. The fact that `.Fits()` was called at all was a Manager specific issue that should be alleviated by the fact that `Unlimited` is now called in `.Fits()`.